R/zchunk_L222.en_transformation.R

```diff
@@ -81,248 +87,540 @@ module_energy_L222.en_transformation_DISABLED <- function(command, ...) {
+    }
+
+    # L222.GlobalTechProfitShutdown_en: Global tech profit shutdown decider and parameters
+    if (any(!is.na(L222.globaltech_retirement$median.shutdown.point))){
+      L222.globaltech_retirement %>%
+        filter(!is.na(L222.globaltech_retirement$median.shutdown.point)) %>%
+        select(sector.name, subsector.name, technology, median.shutdown.point, profit.shutdown.steepness) ->
+        L222.GlobalTechProfitShutdown_en
+    }
+
+    #2d. Calibration and region-specific data
+    #  generate base year calibrated outputs of gas processing by interpolating from historical values
+    L122.out_EJ_R_gasproc_F_Yh %>%
+      complete(nesting(GCAM_region_ID, sector, fuel), year = c(year, BASE_YEARS)) %>%
+      arrange(GCAM_region_ID, year) %>%
+      group_by(GCAM_region_ID, sector, fuel) %>%
+      mutate(value = approx_fun(as.numeric(year), value)) %>%
+      ungroup() %>%
+      filter(year %in% BASE_YEARS) %>%
+      # append region names
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") ->
+      L222.out_EJ_R_gasproc_F_Yh_base
+
+    # append matching calibrated technology sector/subsector/technology to calibrated outputs of gas processing
+    calibrated_techs %>%
+      semi_join(L222.out_EJ_R_gasproc_F_Yh_base, by = c("sector", "fuel")) %>%
+      select(sector, fuel, supplysector, subsector, technology) %>%
+      left_join(L222.out_EJ_R_gasproc_F_Yh_base, by = c("sector", "fuel")) %>%
+      rename(stub.technology = technology) ->
+      L222.out_EJ_R_gasproc_F_Yh
+
+    # L222.StubTechProd_gasproc: calibrated output of gas processing technologies -- writes to all regions, adds calibrated tech info
+    A22.globaltech_coef %>%
+      filter(supplysector == "gas processing") %>%
+      write_to_all_regions(c("region", "supplysector", "subsector", "technology", "minicam.energy.input"), GCAM_region_names) %>%
+      rename(stub.technology = technology) %>%
+      repeat_add_columns(tibble("year" = BASE_YEARS)) %>%
+      left_join_error_no_match(L222.out_EJ_R_gasproc_F_Yh, by = c("region", "supplysector", "subsector", "stub.technology", "year")) %>%
+      # rounds outputs and adds year column for shareweights
+      mutate(calOutputValue = round(value, energy.DIGITS_CALOUTPUT), year.share.weight = year) %>%
+      select(-sector, -GCAM_region_ID, -fuel, -value, -minicam.energy.input) %>%
+      # sets shareweight to 1 if output exists, otherwise 0
+      mutate(share.weight = if_else(calOutputValue > 0, 1, 0)) %>%
+      mutate(subs.share.weight = share.weight) ->
+      L222.StubTechProd_gasproc
+
+    # THE FOLLOWING IS IN THE ORIGINAL BUT ALREADY OMITS THOSE REGIONS WITH NO COAL GAS, SO THERE ARE NO MISSING VALUES, JUST NO ROWS
+    # IS THIS A MISTAKE? NOT SURE IF WE NEED THOSE ROWS. PASSES ALL CHECKS.
+    # "#Coal to gas isn't included in all regions so replace missing values
+    # L222.StubTechProd_gasproc$calOutputValue[ is.na( L222.StubTechProd_gasproc$calOutputValue ) ] <- 0"
+
```

Comment:
I guess I am not sure what this comment is saying.  ARe the zero coal to gas regions just not listed in the input?  Or are they stripped out elsewhere?  Or, does code that follows this not require that all regions be included so that assigning the zeroes isn't necessary?  It's clear that I don't know, so explain what you can, and see if you need that step in or can delete it.

Reply:
@mwisepnnl Yes, `fuel_energy_input` is used by quite a few other chunks: L222, L223, L224, L226, L232, L242, L252.

Reply:
Okay.  Then I guess I am waiting on just the coal to gas clarification.

Reply:
The short version is that there are no NAs, so it doesn't do anything. The long version is that it also doesn't do anything in the old data system which made me concerned that I was missing the point. I did a bit more looking and I think what's happening is that the zeroes are being generated upstream in L122, so this line is entirely redundant. Not sure why it existed in the first place though. I'll delete the comment.